test/functional/p2p_blocksonly.py

```diff
@@ -100,6 +105,13 @@ def blocks_relay_conn_tests(self):
         conn.sync_send_with_ping()
         assert(int(txid, 16) not in conn.get_invs())
 
+    def check_p2p_inv_violation(self, peer):
```

Comment:
This doesn't need to be done here, but if there is a new function for this anyway, it could also be used in the first part for block-relay-only to remove some (almost) duplicate code.